apps/prairielearn/elements/pl-file-editor/pl-file-editor.py

```diff
@@ -150,27 +151,22 @@ def render(element_html, data):
     else:
         html_params["current_file_contents"] = html_params["original_file_contents"]
 
-    if data["panel"] == "question":
-        html_params["question"] = True
-        with open("pl-file-editor.mustache", "r", encoding="utf-8") as f:
-            html = chevron.render(f, html_params).strip()
-    else:
-        html = ""
-
-    return html
+    with open("pl-file-editor.mustache", "r", encoding="utf-8") as f:
+        return chevron.render(f, html_params).strip()
 
 
-def parse(element_html, data):
+def parse(element_html: str, data: pl.QuestionData) -> None:
     element = lxml.html.fragment_fromstring(element_html)
     file_name = pl.get_string_attrib(element, "file-name", "")
     answer_name = get_answer_name(file_name)
     normalize_to_ascii = pl.get_boolean_attrib(
         element, "normalize-to-ascii", NORMALIZE_TO_ASCII_DEFAULT
     )
+    allow_blank = pl.get_boolean_attrib(element, "allow-blank", ALLOW_BLANK_DEFAULT)
 
     # Get submitted answer or return parse_error if it does not exist
-    file_contents = data["submitted_answers"].get(answer_name, None)
-    if not file_contents:
+    file_contents = data["submitted_answers"].get(answer_name, "")
```

Comment:
Changing the default to `""` was necessary for type-checking. I believe this is safe: if `allow-blank` is false, we'll just add an error and ignore the contents entirely. And if `allow-blank` is true, then we don't really care about the distinction between "not present" and "empty".